core/src/test/java/org/apache/struts2/interceptor/FetchMetadataInterceptorTest.java

```diff
@@ -102,22 +107,179 @@ public void testPathInExemptedPaths() throws Exception {
     public void testPathNotInExemptedPaths() throws Exception {
         MockActionInvocation mai = new MockActionInvocation();
         MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
 
         request.addHeader("sec-fetch-site", "foo");
         request.setContextPath("/foobar");
 
         ServletActionContext.setRequest(request);
+        ServletActionContext.setResponse(response);
         ActionContext context = ServletActionContext.getActionContext();
         mai.setInvocationContext(context);
 
         assertEquals("Expected interceptor to NOT accept this request", "403", interceptor.intercept(mai));
     }
 
+    public void testVaryHeaderAcceptedReq() throws Exception {
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+
+        request.addHeader("sec-fetch-site", "foo");
+        request.setContextPath("/foo");
+
+        ServletActionContext.setRequest(request);
+        ServletActionContext.setResponse(response);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        interceptor.intercept(mai);
+
+        assertTrue("Expected vary header to be included", response.containsHeader(ResourceIsolationPolicy.VARY_HEADER));
+        assertEquals("Expected different vary header value", response.getHeader(ResourceIsolationPolicy.VARY_HEADER), FetchMetadataInterceptor.VARY_HEADER_VALUE);
```

Comment:
Let's add static imports for everything under `ResourceIsolationPolicy` and `FetchMetadataInterceptor` please